cliquet/authentication.py

```diff
@@ -79,8 +80,28 @@ def __init__(self, config, realm='Realm'):
         self.cache = oauth_cache
 
     def unauthenticated_userid(self, request):
-        user_id = self._get_credentials(request)
-        return user_id
+        profile = self._get_credentials(request)
+        if profile is None:
+            return None
+
+        # Attach scopes to request.
+        setattr(request, '_fxa_scopes', profile['scope'])
+
+        # Return user-id with prefix.
+        return 'fxa_%s' % profile['user']
+
+    def effective_principals(self, request):
+        """Provide OAuth scopes in principals, in order to allow authorizations
+        policies to rely on them.
+        """
+        principals = super(Oauth2AuthenticationPolicy,
+                           self).effective_principals(request)
+
+        # unauthenticated_userid() is always called before this.
+        scopes = getattr(request, '_fxa_scopes', [])
```

Comment:
Also we will use scope to reduce the permission of a given token.

So if Bob have got ``storage:blog:articles:read`` scope we want to restreint the permission for this token to read on the ``articles`` collection of the ``blog`` bucket.